### PR TITLE
Update IteratorTester example with a greater "steps" value

### PR DESCRIPTION
**Pull request by jbduncan**
Builds upon my earlier work in https://github.com/google/guava/pull/5272 and https://github.com/google/guava/pull/5276, whilst also addressing https://github.com/google/guava/issues/5254 a bit.

### CODE REVIEW

File: guava-testlib/src/com/google/common/collect/testing/IteratorTester.java [hunk 1]
```diff
@@ -50,6 +50,10 @@
  * verify() method, which is called <em>after</em> each sequence and is guaranteed to be called
  * using the latest values obtained from {@link IteratorTester#newTargetIterator()}.
  *
+ * <p>The value you pass to the parameter {@code steps} should be greater than the length of your
+ * iterator, so that this class can check that your iterator exhibits the same edge-case behavior as
```
Comment (cpovirk) [inline guava-testlib/src/com/google/common/collect/testing/IteratorTester.java:54]: What do you think of just saying "...that your iterator exhibits the correct behavior when it is exhaused?"

Comment (jbduncan) [reply guava-testlib/src/com/google/common/collect/testing/IteratorTester.java:54]: Yep, that sounds good to me. :+1:

Comment (jbduncan) [reply guava-testlib/src/com/google/common/collect/testing/IteratorTester.java:54]: I'm not sure if you were expecting me to update the docs or if you were happy to do it yourself?

Comment (cpovirk) [reply guava-testlib/src/com/google/common/collect/testing/IteratorTester.java:54]: Sorry, I'm taking care of it. The PR is out for internal review.

Comment (jbduncan) [reply guava-testlib/src/com/google/common/collect/testing/IteratorTester.java:54]: Cheers @cpovirk! 👍

**Pull request closed without merging**